src/main.cpp

```diff
@@ -316,8 +316,9 @@ void generate_piece_moves(Move *const movelist,
     const BB all = pos.colour[0] | pos.colour[1];
     const BB to_mask = only_captures ? pos.colour[1] : ~pos.colour[0];
     const BB pawns = pos.colour[0] & pos.pieces[Pawn];
+    generate_pawn_moves(
+        movelist, num_moves, north(pawns) & ~all & (only_captures ? 0xFF00000000000000ULL : 0xFFFFFFFFFFFF0000ULL), -8);
```

Comment:
The second mask can be written more compact as ```~0```.

Reply:
i was looking for a way to have a more compact mask and i couldn't find one, thank you